### PR TITLE
Support all content#393

### LINKED ISSUE #393: Ensure proxy can work for all content types
Not just JSON. It should also allow for:
* Images
* Javascript
* HTML
* CSS

Essentially, we want to test and ensure that it is working perfectly for proxying a web server with general content.

In cases where logging is enabled via the proxy, it should **only** still capture JSON though.

### PR DESCRIPTION
**Pull request by jperata**
Closes #393 
 WIP
- Change to redirection added.

Missing
-  Had a change to pass from string to buffer across client to avoid issues with encoding and get the "pure" payload, but it's reverted here since it was failing on server side (node was being dropped).

Will retake after work on virtual-device

### CODE REVIEW

File: lib/client/tcp-client.ts [hunk 1]
```diff
@@ -28,14 +29,77 @@ export class TCPClient {
...
-            client.write(data);
+            client.write(requestData);
         });
 
 
         // Add a 'data' event handler for the client socket
         // data is what the server sent to this socket
-        client.on("data", function(data: Buffer) {
-            callback(data, null, null);
+        client.on("data", function(incommingData: Buffer) {
```
Comment (jkelvie) [inline lib/client/tcp-client.ts]: Almost all the handling done here is actually in the httpBuffer class, which the data coming back is pushed into on the callback:
https://github.com/bespoken/bst/blob/7701d87e3dc07554f4c7d2c1123667a31ec7d90f/lib/client/bespoke-client.ts#L106

In that class, I pull out the response code, method, path, headers, and other data from the payload automatically.

For a 302, why do we need to do additional processing? What is the exact payload that comes back, and why does it not work?

Comment (jperata) [reply lib/client/tcp-client.ts]: On a 302 , we just get the header with the new Location, so if we don't resend the data to that one, we end loading forever
Ex.
bst  proxy http --bstHost localhost --bstPort 5000 --targetDomain bit.ly 80

Comment (jkelvie) [reply lib/client/tcp-client.ts]: The targetDomain should not change. The reply should do the redirect though.

I believe the problem is here:
https://github.com/bespoken/spokes/blob/master/lib/server/HTTPPipe.ts#L61

We are not setting the statusCode or statusMessage on the sourceResponse:
https://nodejs.org/api/http.html#http_response_statusmessage
https://nodejs.org/api/http.html#http_response_statuscode

Both of these should be set from the values on the response.

Comment (jkelvie) [reply lib/client/tcp-client.ts]: I've also attached a simple project to test with:
[redirect-test.zip](https://github.com/bespoken/bst/files/1455502/redirect-test.zip)

**Pull request closed without merging**